### ISSUE TITLE
Conversation Is Not Saved

### ISSUE THREAD

**Issue by mhendy25** [adrianhajdin/project_openai_codex#60]
Does anyone have an idea why the chat doesn't save the previous prompts and their responses (see picture below)? Do I have to manually save the entire chat history? Thanks!
<img width="1440" alt="Screen Shot 2023-06-23 at 8 57 46 PM" src="https://github.com/adrianhajdin/project_openai_codex/assets/88539544/51e640bd-2df5-4d0d-8ef2-837a1b3cf815">

**Comment by tydolla00** [comment]
yes you have to say the whole history. [...prevMessages, newMessage]. You should be sending the array in the body of the request.

**Comment by justtfun** [comment]
@Kritikavig facing same error 429 issue....please help!
